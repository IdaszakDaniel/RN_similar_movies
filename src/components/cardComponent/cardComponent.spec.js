import React from 'react'
import { render } from 'react-native-testing-library'
import CardComponent from './cardComponent'

describe('Card Component', () => {
  it('renders the image with the test URI', () => {
    const props = {
      data: {
        poster_path: 'test'
      }
    }

    const tree = render(<CardComponent {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the image with proper styles', () => {
    const props = {
      data: {
        poster_path: 'test'
      }
    }

    const rendered = render(<CardComponent {...props} />)
    const wrapperComponent = rendered.getByTestId('image')

    expect(wrapperComponent.props.style).toMatchObject({
      width: '100%',
      height: '100%'
    })
    expect(wrapperComponent.props.imageStyle).toMatchObject({
      borderRadius: 10
    })
  })
})
