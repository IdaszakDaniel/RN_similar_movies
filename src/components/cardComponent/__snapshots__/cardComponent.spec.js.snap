// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the image with the test URI 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://image.tmdb.org/t/p/w185/test",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          Object {
            "borderRadius": 10,
          },
        ]
      }
      testID="image"
    />
  </View>
</View>
`;
